config/config/src/test/java/io/helidon/config/ConfigMappersTest.java

```diff
@@ -256,4 +350,26 @@ private Config createConfig() {
                 .build();
     }
 
+    public enum MyType {A, B, SPECIAL_B};
+
+    public enum ConfusingType {RED, Red, BLUE};
+
+    private Config createEnumConfig() {
+        return Config.just(ConfigSources.create(Map.of(
```

Comment:
"type", "a"
 - or -
"type", "SPECIAL_B"

^^^
This is the prime use case IMHO.

"type" is of type MyType, and the value will be an enum value of MyType.

Reply:
Again, I'm sorry I seem to be missing your point.

In particular, I'm not getting what role you see the string "type" playing in the conversion/mapping.

Let's use the color example (slightly expanded but not essentially different) from the doc instead of the artificial and (apparently) confusing config keys from the test.

```java
enum Color {RED, YELLOW, BLUE_GREEN};
```

```yaml
myapp:
    house:
      size: 2500
      tint: "blue-green"
    car:
      color: "Red"
      make: "Honda"
      model: "Pilot"
    warning: "YELLOW"
```
As with any config value of any type, there's nothing in the config key or value that conveys the type to which the config value might be mapped. That's up entirely to the application and its use of `config.asInt()` or `config.as(SomeType.class)` or the equivalent. 

To convert the config value with key `myapp.house.tint` to `Color` the app could use
```
   Color houseTint = config.get("myapp.house.tint").as(Color.class).get();
```

Can you recast your comment using this example so I can better understand what you're getting at?

Reply:
sorry, this comment was more of a continuation of the previous one - we should just be testing the simpler prototypical usage where the ConfigValue (e.g., "A" or "SPECIAL_B") is being carried in a config key named "type", rather than mangled to use camel case or hyphens, etc.